extensions/Worker.Extensions.SignalRService/src/Configuration/ServiceManagerOptionsSetup.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.SignalR.Management;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.Functions.Worker.SignalRService
+{
+    internal class ServiceManagerOptionsSetup
+    {
+        private readonly IConfiguration _configuration;
+        private readonly AzureComponentFactory _azureComponentFactory;
+        private readonly IEnumerable<IConfigureOptions<ServiceManagerOptions>> _configureOptions;
+
+        public ServiceManagerOptionsSetup(IConfiguration configuration, AzureComponentFactory azureComponentFactory, IEnumerable<IConfigureOptions<ServiceManagerOptions>> configureOptions)
```

Comment:
You shouldn't need to import `IConfigureOptions` yourself. You import `IOptionsMonitor<ServiceManagerOptions>`, and the applicable configurations will be ran for you when retrieving the options object.

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-7.0

This is for aspnetcore, but applies to generic host as well.

Reply:
`IConfigureOptions` is used instead of `IOptionsMonitor` because `ServiceManagerOptions` is not a typical options structure. It has a method called `UseJsonObjectSerializer(ObjectSerializer)`, where the object serializer passed in the parameter is recorded in an internal field and we cannot get it back from outside. Therefore, I have to import the `ICongureOptions` myself.